File: app/src/main/java/uk/co/agfsoft/skyweather/presenter/FavouritesPresenter.java
package uk.co.agfsoft.skyweather.presenter;

import java.util.List;

import javax.inject.Inject;

import uk.co.agfsoft.skyweather.model.Model;
import uk.co.agfsoft.skyweather.model.WeatherCity;
import uk.co.agfsoft.skyweather.view.FavouritesView;



public class FavouritesPresenter {
    private final FavouritesView favouritesView;
    private final Model model;

    @Inject
    public FavouritesPresenter(FavouritesView favouritesView, Model model) {
        this.favouritesView = favouritesView;
        this.model = model;
    }

    public void addFavouriteClicked() {
        favouritesView.showAddFavouriteView();
    }

    public void viewReady() {
        List<WeatherCity> favourites = model.getFavourites();
        favouritesView.showIntroOverlay(favourites.size() == 0);
        favouritesView.updateFavourites(favourites);
    }

    public List<WeatherCity> getFavourites() {
        return model.getFavourites();
    }

    public void favouriteClicked(WeatherCity weatherCity) {
        favouritesView.showWeatherForCity(weatherCity);
    }
}

File: app/src/main/kotlin/uk/co/agfsoft/skyweather/presenter/FavouritesPresenter.kt
package uk.co.agfsoft.skyweather.presenter

import javax.inject.Inject

import model.Model
import model.WeatherCity
import uk.co.agfsoft.skyweather.view.FavouritesView


class FavouritesPresenter @Inject
constructor(private val favouritesView: FavouritesView, private val model: Model) {

    val favourites: List<WeatherCity>
        get() = model.favourites

    fun addFavouriteClicked() {
        favouritesView.showAddFavouriteView()
    }

    fun viewReady() {
        val favourites = model.favourites
        favouritesView.showIntroOverlay(favourites.size == 0)
        favouritesView.updateFavourites(favourites)
    }

    fun favouriteClicked(weatherCity: WeatherCity) {
        favouritesView.showWeatherForCity(weatherCity)
    }
}
